src/main/java/com/atomicjar/logging/LogFmtMarker.java

```diff
@@ -26,13 +26,14 @@ public class LogFmtMarker implements Marker {
 
     private final Map<String, Object> data;
 
-    private LogFmtMarker(Map<String, Object> data) {
-        this.data = Collections.unmodifiableMap(data);
+    private LogFmtMarker(Map<String,Object> data) {
+        this.data = data;
     }
 
 
     public static LogFmtMarker with(String key, Object value) {
-        return new LogFmtMarker(Map.of(key, value));
+        var marker = new LogFmtMarker(Collections.emptyMap());
+        return value == null ? marker : marker.and(key, value);
```

Comment:
nit: something like `new LogFmtMarker(value == null ? Map.of() : Map.of(key, value))` would be easier to follow and do less allocations

Reply:
done